### ISSUE TITLE
[Docker] stay on specific tag and update digest only

### ISSUE THREAD

**Issue by zhangyoufu** [dependabot/dependabot-core#1971]
```
FROM registry:2@sha256:3a8eef8d0a818b9bbb4bd17667253473e2d99935ccbbd37649af6bcaa064cf0d
```

I think that dependabot will upgrade base image tag for me once there is `registry:3` available.

I want to stay at registry:2 and update digest only. Is there any configuration available for this case?

**Comment by deivid-rodriguez** [comment]
I think the `version-update:ignore-major` ignore configuration should be the right option for this. I expect #6115 to get that working 💪.

**Comment by lorengordon** [comment]
Thanks @deivid-rodriguez! Would it be possible to get an example config of how to use this? Or a link to the right doc? Just to close the loop...

**Comment by deivid-rodriguez** [comment]
Sure, have a look at https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates#example-disabling-version-updates-for-some-dependencies. You probably want something like

```yaml
version: 2
updates:
  # Configuration for Dockerfile
  - package-ecosystem: "docker"
    directory: "/"
    schedule:
      interval: "weekly"

    ignore:
      # For all images, ignore all major updates
      - dependency-name: "*"
        update-types: ["version-update:semver-major"]
```

**Comment by lorengordon** [comment]
I saw that, but nothing that seems to really explain what `dependency-name` is for every ecosystem. What does the `dependency-name` match in a Dockerfile? For example, how might I constrain this to just python 3.10? Is it the exact FROM line?

```
FROM python:3.10.9-buster
```

```
version: 2
updates:
  # Configuration for Dockerfile
  - package-ecosystem: "docker"
    directory: "/"
    schedule:
      interval: "weekly"

    ignore:
      # For all images, ignore all major updates
      - dependency-name: "python:3.10.9-buster"
        update-types: ["version-update:semver-major", "version-update:semver-minor"]
```

**Comment by deivid-rodriguez** [comment]
For the Docker ecosystem, "dependency name" = "image name", and "dependency version" = "tag version", so you want just "python" in there. What's on the other side of the `:` is the version.

**Comment by lorengordon** [comment]
Hmm, ok, that's tricky for multi-stage builds. I could have several images with the same name but different tags in the same Dockerfile.

**Comment by yob** [comment]
This sounds promising, but I'm struggling to get a digest-only update 🤔

I have a Dockerfile with the following base:
```
FROM public.ecr.aws/docker/library/ruby:2.7.6-buster@sha256:3ee0c8dba7051b2974ba5bead0f71db7f3f17e9302036182dfcbb401cff7ddd8 AS base
```

My goal is to get a PR that keeps the same version of ruby and updates only the digest to the most recent:
```
FROM public.ecr.aws/docker/library/ruby:2.7.6-buster@sha256:f2ba73c31b11be325121bc5018231e21e13348c3b3bc75be44e7a6568893f5c1
```

I tried a few config options.

The first two allowed dependabot to open a PR that bumped the image tag from `2.7.6-buster` to `2.7.7-buster`:
```
    ignore:
      # This should mean we auto update the image digest but never the tag
      - dependency-name: "*"
        update-types: ["version-update:semver-major"]
```
```
    ignore:
      # This should mean we auto update the image digest but never the tag
      - dependency-name: "*"
        update-types: ["version-update:semver-major", "version-update:semver-minor"]
```

The final one permitted no updates:
```
    ignore:
      # This should mean we auto update the image digest but never the tag
      - dependency-name: "*"
        update-types: ["version-update:semver-major", "version-update:semver-minor", "version-update:semver-patch"]
```
output from the final run:
```
pdater | INFO <job_600356981> Checking if docker/library/ruby 2.7.6-buster needs updating
updater | INFO <job_600356981> Ignored versions:
updater | INFO <job_600356981>   version-update:semver-major - from .github/dependabot.yml
updater | INFO <job_600356981>   version-update:semver-minor - from .github/dependabot.yml
updater | INFO <job_600356981>   version-update:semver-patch - from .github/dependabot.yml
  proxy | 2023/02/09 03:33:57 [014] GET https://public.ecr.aws:443/v2/docker/library/ruby/tags/list
  proxy | 2023/02/09 03:33:57 [014] 401 https://public.ecr.aws:443/v2/docker/library/ruby/tags/list
  proxy | 2023/02/09 03:33:57 [016] GET https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:57 [016] 200 https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:57 [018] GET https://public.ecr.aws:443/v2/docker/library/ruby/tags/list
  proxy | 2023/02/09 03:33:57 [018] 200 https://public.ecr.aws:443/v2/docker/library/ruby/tags/list
  proxy | 2023/02/09 03:33:57 [020] HEAD https://public.ecr.aws:443/v2/docker/library/ruby/manifests/latest
  proxy | 2023/02/09 03:33:57 [020] 401 https://public.ecr.aws:443/v2/docker/library/ruby/manifests/latest
  proxy | 2023/02/09 03:33:57 [022] GET https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:57 [022] 200 https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:57 [024] HEAD https://public.ecr.aws:443/v2/docker/library/ruby/manifests/latest
  proxy | 2023/02/09 03:33:57 [024] 200 https://public.ecr.aws:443/v2/docker/library/ruby/manifests/latest
  proxy | 2023/02/09 03:33:58 [026] HEAD https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2.1
  proxy | 2023/02/09 03:33:58 [026] 401 https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2.1
  proxy | 2023/02/09 03:33:58 [028] GET https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:58 [028] 200 https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:58 [030] HEAD https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2.1
  proxy | 2023/02/09 03:33:58 [030] 200 https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2.1
  proxy | 2023/02/09 03:33:58 [032] HEAD https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2.0
  proxy | 2023/02/09 03:33:58 [032] 401 https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2.0
  proxy | 2023/02/09 03:33:58 [034] GET https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:58 [034] 200 https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:58 [036] HEAD https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2.0
  proxy | 2023/02/09 03:33:58 [036] 200 https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2.0
  proxy | 2023/02/09 03:33:58 [038] HEAD https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2
  proxy | 2023/02/09 03:33:58 [038] 401 https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2
  proxy | 2023/02/09 03:33:58 [040] GET https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:58 [040] 200 https://public.ecr.aws:443/token/?service=public.ecr.aws&scope=aws
  proxy | 2023/02/09 03:33:58 [042] HEAD https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2
  proxy | 2023/02/09 03:33:58 [042] 200 https://public.ecr.aws:443/v2/docker/library/ruby/manifests/3.2
updater | INFO <job_600356981> All updates for docker/library/ruby were ignored
updater | INFO <job_600356981> Finished job processing
```

Is there a config that will allow this image to update the digest only, and leave the tag on `2.7.6-buster`?

**Comment by deivid-rodriguez** [comment]
Interesting. I made a similar test [here](https://github.com/dsp-testing/dependabot-core-5758) and I got a proper update bumping only the digest.

Judging from the difference between my test case and yours, I guess in my case the digest is being considered as the "patch component" of the version (since the Dockerfile was only pinning to `<major>.<minor?`), and that's why that one worked?

I'll reopen and try to have a closer look at your case.
